ml-algorithms/src/main/java/org/opensearch/ml/engine/ModelHelper.java

```diff
@@ -137,6 +138,46 @@ public void downloadPrebuiltModelConfig(String taskId, MLRegisterModelInput regi
         }
     }
 
+    public boolean isModelAllowed(MLRegisterModelInput registerModelInput, List modelMetaList) {
+        String modelName = registerModelInput.getModelName();
+        String version = registerModelInput.getVersion();
+        MLModelFormat modelFormat = registerModelInput.getModelFormat();
+        for (Object meta: modelMetaList) {
+            String name = (String) ((Map<String, Object>)meta).get("name");
+            List<String> versions = (List) ((Map<String, Object>)meta).get("version");
+            List<String> formats = (List) ((Map<String, Object>)meta).get("format");
+            if (name.equals(modelName) && versions.contains(version.toLowerCase(Locale.ROOT)) && formats.contains(modelFormat.toString().toLowerCase(Locale.ROOT))) {
+                return true;
+            }
+        }
+        return false;
+    }
+
+    public List downloadPrebuiltModelMetaList(String taskId, MLRegisterModelInput registerModelInput) throws PrivilegedActionException {
+        String modelName = registerModelInput.getModelName();
+        String version = registerModelInput.getVersion();
+        try {
+            return AccessController.doPrivileged((PrivilegedExceptionAction<List>) () -> {
+
+                Path registerModelPath = mlEngine.getRegisterModelPath(taskId, modelName, version);
+                String cacheFilePath = registerModelPath.resolve("model_meta_list.json").toString();
+                String modelMetaListUrl = mlEngine.getPrebuiltModelMetaListPath();
+                DownloadUtils.download(modelMetaListUrl, cacheFilePath, new ProgressBar());
+
+                List<?> config = null;
+                try (JsonReader reader = new JsonReader(new FileReader(cacheFilePath))) {
+                    config = gson.fromJson(reader, List.class);
+                }
+
+                return config;
+            });
+        } catch (Exception e) {
+            throw e;
```

Comment:
If we just throw the exception directly, seems no need to add `catch (Exception e)`

Reply:
Good point, removed it.